atlasdb-cli/src/main/java/com/palantir/atlasdb/cli/command/TruncatePaxosLog.java

```diff
@@ -0,0 +1,103 @@
+        LeaderConfig leaderConfig;
+        try {
+            leaderConfig = getServiceConfigModule().provideAtlasDbConfig().leader().get();
+        } catch (IllegalStateException e) {
+            System.err.println("Error: Config file is missing required leader block configuration.");
+            return 1;
+        }
+
+        // ensure acceptor log is empty
+        PaxosStateLog<PaxosAcceptorState> paxosLog = new PaxosStateLogImpl<PaxosAcceptorState>(leaderConfig.acceptorLogDir().toString());
+        long greatestLogEntry = paxosLog.getGreatestLogEntry();
+        if (greatestLogEntry > PaxosAcceptor.NO_LOG_ENTRY) {
+            System.err.println("Error: This log has been written to already, but this cli was designed to "
+                    + "be ran against a new server that has never been started.");
+            return 1;
+        }
+
+        // create paxos service
+        PaxosLeaderElectionService leaderElectionService;
+        try {
+            leaderElectionService = (PaxosLeaderElectionService) Leaders.create(
+                    getServiceConfigModule().provideServicesConfig().sslSocketFactory(),
+                    resource -> {},
+                    leaderConfig);
+        } catch (ClassCastException e) { //impossible currently as Paxos is only existing implementation
+            System.err.println("Error: LeaderElectionService is not a Paxos implementation, which this cli is designed for.");
+            return 1;
+        }
+
+        // get the greatest log according to other leaders
+        int quorumSize = leaderConfig.quorumSize();
+        ImmutableList<PaxosAcceptor> acceptors = leaderElectionService.getAcceptors();
+        long maxGreatestLog = -1L;
+        int numSuccesses = 0;
+        for (PaxosAcceptor paxosAcceptor : acceptors) {
+            try {
+                long lastLogEntry = paxosAcceptor.getLatestSequencePreparedOrAccepted();
+                maxGreatestLog = Math.max(maxGreatestLog, lastLogEntry);
+                numSuccesses++;
+            } catch (Exception e) {
+                System.out.println("Warn: Failed to get last log entry from: " + paxosAcceptor);
+            }
+        }
+        if (numSuccesses < quorumSize) {
+            System.err.println("Error: Failed because we could not talk to quorum servers to truncate this log correctly."
+                    + "  This error has likely occurred because you are recovering from having lost at least quorum number"
+                    + " of servers.  In this case you should take all services down and run the deletePaxosLog CLI against each of them.");
+            return 1;
+        }
+
+        // never accept any of the pre-existing promises; truncate
+        PaxosAcceptorState neverAcceptState = PaxosAcceptorState.newState(new PaxosProposalId(Long.MAX_VALUE, UUID.randomUUID().toString()));
+        paxosLog.writeRound(maxGreatestLog + 1, neverAcceptState);
+        paxosLog.truncate(maxGreatestLog);
```

Comment:
So it depends on whether or not line 49 return PaxosAcceptor.NO_LOG_ENTRY because there are no files or because there is actually a file for -1.  In the former the truncate is needed to do the verify at line 93, in the latter the truncate isn't needed.

Presumably if we're running this CLI it's because there are no files and so the truncate isn't needed?  Do we do anything like auto-write a NO_LOG_ENTRY file on init?  Because then we're at the latter case and need to truncate.

The truncate is strictly safe, and we could also just remove the if check after it then.